include/boost/geometry/algorithms/detail/buffer/buffered_piece_collection.hpp

```diff
@@ -409,13 +399,10 @@ struct buffered_piece_collection
     inline void update_turn_administration()
     {
         std::size_t index = 0;
-        for (typename boost::range_iterator<turn_vector_type>::type it =
-            boost::begin(m_turns); it != boost::end(m_turns); ++it, ++index)
+        for (auto& turn : m_turns)
         {
-            buffer_turn_info_type& turn = *it;
-
             // Update member used
-            turn.turn_index = index;
+            turn.turn_index = index++;
```

Comment:
`for_each_with_index` is used below. Would it make sense to use it here as well?

Reply:
It's currently a const version only.
But I can add an overload.

Reply:
:heavy_check_mark: